srml/support/src/dispatch.rs

```diff
@@ -317,23 +317,11 @@ macro_rules! decl_module {
 		{ $( $constants:tt )* }
 		[ $( $dispatchables:tt )* ]
 		$(#[doc = $doc_attr:tt])*
-		$vis:vis fn deposit_event $(<$dpeg:ident $(, $dpeg_instance:ident)?>)* (
-			$($param_name:ident : $param:ty),*
-		) { $( $impl:tt )* }
+		$vis:vis fn deposit_event
 		$($rest:tt)*
 	) => {
-		$crate::decl_module!(@normalize
-			$(#[$attr])*
-			pub struct $mod_type<$trait_instance: $trait_name$(<I>, I: $instantiable $(= $module_default_instance)?)?>
-			for enum $call_type where origin: $origin_type, system = $system
-			{ $( $other_where_bounds )* }
-			{ $vis fn deposit_event $(<$dpeg $(, $dpeg_instance)?>)* ($( $param_name: $param ),* ) { $( $impl )* } }
-			{ $( $on_initialize )* }
-			{ $( $on_finalize )* }
-			{ $( $offchain )* }
-			{ $( $constants )* }
-			[ $( $dispatchables )* ]
-			$($rest)*
+		compile_error!(
+			"`deposit_event` function is reserved and must follow the syntax: `$vis:vis fn deposit_event() = default;`"
```

Comment:
I think we should delegate this to the `__check_reserved_fn_name` macro.

Reply:
actually check_reserved_fn_name check if some dispatchable function have the name of a reserved function. Like someone can declare a dispatchable function on_finalize just by not following on_finalize accepting signature and this is confusing.

Thus if we remove this branch and I add an old syntax `deposit_event<T>() =default;` somewhere the error is actually:

```
error: no rules expected the token `normalize`
   --> srml/balances/src/lib.rs:399:1
    |
399 | / decl_module! {
400 | |     pub struct Module<T: Trait<I>, I: Instance = DefaultInstance> for enum Call where origin: T::Origin {
401 | |         /// The minimum amount required to keep an account open.
402 | |         const ExistentialDeposit: T::Balance = T::ExistentialDeposit::get();
...   |
506 | |     }
507 | | }
    | |_^ no rules expected this token in macro call
    |
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

```

note that those parsing message error should be easier to solve with procedural macro and I'm currently working on porting it to this.

Reply:
Yeah okay, makes sense.